vendor/engines/split_accounts/spec/engine_helper.rb

```diff
@@ -0,0 +1,9 @@
+# Rspec configurations for this engine
+RSpec.configure do |config|
+  
+  # Skip split_accounts tests if split_accounts feature is disabled
+  unless SettingsHelper.feature_on?(:split_accounts)
```

Comment:
Your point about the class variables not resetting makes sense. I'm not so worried if the only places you use this metadata are only within the engine.